rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -167,11 +164,115 @@ void getfirstInputSighHash_p2sh_pegin_v1() {
 
         // Act
         Optional<Sha256Hash> firstInputSigHash = BitcoinUtils.getFirstInputSigHash(btcTx);
+
         // Assert
         assertTrue(firstInputSigHash.isPresent());
         assertTrue(isSigHashValid(firstInputSigHash.get(), pubKeys, signatures));
     }
 
+    @Test
+    void extractRedeemScriptFromInput_forP2shP2wshTx_withInvalidRedeemScript_shouldReturnEmpty() {
```

Comment:
I think we are missing some test cases for the `extractRedeemScriptFromInput` segwit method. For instance. extracting the redeem script from a:

- segwit pegout
- segwit pegout multiples ins & out
- segwit pegout with no change for the fed
- segwit migration
- segwit migration multiples ins & out

Wdyt?

Reply:
https://github.com/rsksmart/rskj/pull/3073